jax/_src/lax/linalg.py

```diff
@@ -321,7 +322,8 @@ def cholesky_jvp_rule(primals, tangents):
   # Forward-mode rule from https://arxiv.org/pdf/1602.07527.pdf
   def phi(X):
     l = jnp.tril(X)
-    return l / (jnp._constant_like(X, 1) + jnp.eye(X.shape[-1], dtype=X.dtype))
+    with jax.numpy_rank_promotion('allow'):
```

Comment:
Is it better to add allow-promotion scopes, or just to make the promotion explicit? Not sure how painful the latter would be.

Reply:
I went back and forth on that – in most of these cases though I judged that doing it manually obscures the logic. For example, here, the manual approach would be something like
```python
return l / (jnp._constant_like(X, 1) + lax.expand_dims(jnp.eye(X.shape[-1], dtype=X.dtype), range(l.ndim - 2)))
```
And in the cases further below, there are multiple such casts that are required.

Reply:
I gave it a shot - making promotion explicit isn't too bad. Thanks for the nudge!